isis/src/control/objs/ControlNet/ControlNet.cpp

```diff
@@ -552,24 +573,24 @@ namespace Isis {
    */
   void ControlNet::measureDeleted(ControlMeasure *measure) {
     ASSERT(measure);
-
     QString serial = measure->GetCubeSerialNumber();
-    ASSERT(cubeGraphNodes->contains(serial));
-    ControlCubeGraphNode *node = (*cubeGraphNodes)[serial];
+    ASSERT(m_vertexGraph->contains(serial));
 
     // remove connections to and from this node
     if (!measure->IsIgnored() && !measure->Parent()->IsIgnored()) {
       // Break connections
       measureIgnored(measure);
     }
+    
+    // Remove the measure from the associated node.  
+    m_controlGraph[m_vertexMap[serial]].measures.remove(measure->Parent());
 
-    // Remove the measure from the node.  If this caused the node to be empty,
-    // then delete the node.
-    node->removeMeasure(measure);
-    if (!node->getMeasureCount()) {
-      delete node;
-      node = NULL;
-      cubeGraphNodes->remove(serial);
+    // Decrement the edge strength & remove edge if -> 0 happens in measureIgnored
+    // If this caused the node to be empty, then delete the node.
```

Comment:
This is still going to be implemented right? The following code does something else, but we need to do both.

Reply:
We need the code in the measureIgnored method here too.

Reply:
@jessemapel My comment was in the wrong place and did need to be updated. I think `measureIgnored` will always get called inside of `measureDeleted` if the graph needs to be updated and the edge pruning will always take place in there.... but this seems like an odd design, so I should probably look back at what _exactly_ happens to the old graph in these two functions.

Reply:
ah I didn't notice that. I agree that is rather odd functionality.